client/asset/eth/eth.go

```diff
@@ -153,6 +158,9 @@ type ExchangeWallet struct {
 	currentTip *types.Block
 
 	acct *accounts.Account
+
+	lockedFunds    uint64 // gwei
```

Comment:
That wouldn't work because we assign a random nonce to each coin, so coin IDs will all be unique. I think if we support multiple addresses, then we can just do an `address -> uint64` map.

Reply:
> That wouldn't work because we assign a random nonce to each coin, **so coin IDs will all be unique**

That's why it **would** work. Before the nonce, we didn't have anything to key on. Now we do. I like the idea of a `map[string]uint64` or similar, where the key is `hex.EncodeToString(coinID)`

Reply:
Now that I think about it more I guess are some benefits to having the map. I was thinking that there's no reason for the map because on every call to `FundOrder`, a new coin is created, so there would never be a case where two orders are funded by the same coins. However when `ReturnCoins` is called, we could avoid decreasing the locked amount if `ReturnCoins` is called with a coin that is not in the map. This could happen if `ReturnCoins` is called twice with coins from the same order for some reason. I don't think the map really serves a purpose if there are no bugs in the rest of the codebase, but might as well have it as a safety check.